packages/core/contracts/oracle/implementation/OptimisticOracle.sol

```diff
@@ -93,15 +93,17 @@ contract OptimisticOracle is OptimisticOracleInterface, Testable, Lockable {
         uint256 timestamp,
         bytes ancillaryData,
         int256 proposedPrice,
-        uint256 expirationTimestamp
+        uint256 expirationTimestamp,
+        address currency
```

Comment:
This event parameter allows the bot to directly approve any collateral currencies from `ProposePrice` events, which would be used by the bot if for some reason the collateral currency from a proposal was not already approved when it was pre-proposal.

Reply:
Just playing devil's advocate: could we just approve if we decide to dispute and look up the full request in that case? Or does that cause other problems? I only say that because each of these adds about 200 gas to each log message.

Reply:
also if we moved over to a DSProxy model we would be able to do the approval in the same time as the dispute logic and would not need do any pre-approval logic.

However, not really any downside (other than gas) to adding thins IMHO.

Reply:
The reason why I like blanket approving all collateral during the `update` method is so that we can better isolate failures during the `sendDisputes` and `sendProposals` method to the actual propose/dispute logic. If for some reason the approvals fail, then we'd want these failures to happen as early as possible, not right before a dispute needs to be sent.

Reply:
My very weak opinion is that it doesn't seem worth the extra 400 gas (200 for the read and 200 for the extra log field) for a bot optimization -- however, I'm cool either way. LGTM!